app/controllers/submissions_controller.rb

```diff
@@ -31,6 +39,14 @@ def update
       @submission = Submission.find(params[:id])
       @submission.update_columns(admin_id: submission_params[:admin_id])
       redirect_to code_review_submissions_path(@submission.code_review)
+    elsif submission_params['code_review_id']
+      @submission = Submission.find(params[:id])
+      if CodeReview.find_by_id(submission_params[:code_review_id])
+        @submission.update_columns(code_review_id: submission_params[:code_review_id])
+        redirect_to new_submission_review_path(@submission), notice: "Submission reassigned to #{@submission.code_review.course_description_and_code_review_title}"
```

Comment:
Shouldn't this be in the `update` action?

Reply:
It is in the update action, tho I don't love all the if elsif going on there. I'm not sure how else I'd do it tho.

Reply:
Oh ha the Github preview hid a bunch of lines. Why are you using `update_columns` here? I think you'd want to basically just do:

```
@submission = Submission.find(params[:id])
@submission.update(submission_params)
```

And then let the model figure out how to properly update itself.

Then the only `if` logic is for redirecting appropriately.

Reply:
I think the reason I'm using update_columns in that method is because otherwise it triggers the before_save update_times_submitted callback. But maybe better would be to only run the callback if parameters doesn't include some parameter. I'll try playing with that, but if you have a suggestion for a specific way that is good to do that, I'm all ears. Thanks!

Reply:
Oh, and the submissions controller merges in review_status: 'pending' into submission_params. Which I don't want to do in these cases. I'll see about refactoring.